app/models/manageiq/providers/google/resource.rb

```diff
@@ -0,0 +1,40 @@
+module ManageIQ::Providers::Google
+  class Resource
+    def initialize(uri)
+      @type = self.class.parse_resource_type(uri)
+    end
+
+    attr_reader :type
+
+    def snapshot?
+      @type == 'snapshots'
+    end
+
+    def image?
+      @type == 'images'
+    end
+
+    def disk?
+      @type == 'disks'
+    end
+
+    # Parses a Google resource url and returns the resource from the path.
+    # This method understands both complete urls (starting with
+    # https://www.googleapis.com) as well as partials.
+    def self.parse_resource_type(uri)
+      return nil if uri.blank?
+
+      # We can cheat here a bit - most the google APIs (at least in compute.v1)
+      # define resources directly under their type (i.e. snapshots/foo or
+      # images/bar). However, 'images' allow the specification of 'family' which
+      # is an alias to the latest image in a family, breaking the rule.
+      parts = URI.parse(uri).path.split('/')
```

Comment:
just a suggestion, if you would name the parts, someone revisiting this code knows what they mean.
e.g
```ruby
images, family_or_type = parts[-2], parts[-3]
```
as you see, although I read this method now a couple of times, I still dont know the real semantic of those parts 😄